### ISSUE TITLE
question about setting mask for coarse graph

### ISSUE THREAD

**Issue by Chen-Cai-OSU** [szzhang17/Scaling-Up-Graph-Neural-Networks-Via-Graph-Coarsening#1]
Hello,
Thank you for making the code available. I find the here [code](https://github.com/szzhang17/Scaling-Up-Graph-Neural-Networks-Via-Graph-Coarsening/blob/main/GCN/utils.py#L135) a bit confusing
```
new_train_mask = torch.BoolTensor(np.sum(C.dot(train_labels), axis=1))
mix_label = torch.FloatTensor(C.dot(train_labels))
mix_label[mix_label > 0] = 1
mix_mask = torch.sum(mix_label, dim=1)
new_train_mask[mix_mask > 1] = False
```
To my understanding, new_train_mask denotes the new mask for the coarse graph. Why do you use the label of original graph to create the mask? I think `train_label` should be replaced with `H_train_mask`.

**Comment by szzhang17** [comment]
Hi Cai, we use `train_label` because we find that discarding super-nodes with mixed labels often benefits the accuracy. Through `train_label` and `mix_mask`, we remove super-nodes with mixed labels from the training set. If you want to use all the labeled nodes on the coarse graph, you can use `H_train_mask`  to create the new mask.
